### DISCUSSION TITLE
update_focused_file doesn't focus file in another repo

### DISCUSSION THREAD

**Discussion by ckt114** [nvim-tree/nvim-tree.lua#2219] [Q&A]
**Is this a question?**
No

**Can this functionality be implemented utilising API?**
Not sure

**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
With the settings `update_focused_file = { enable = true, update_root = false }` when I open a file in repo A the tree is focused on the folder where the file is located and the root folder is the repo's base, where `.git` folder is. This is working as expected. However, if I open another file in repo B, the tree doesn't update to repo B's base.

I tried setting `update_focused_file.update_root = true` and the tree does follows the file in repo B, however, the root folder is not repo B's base, but the folder deep inside where the file is located. I would like for nvim-tree to always make the repo's base the root folder, not the folder where the file is located as root.

**Describe alternatives you've considered**
I looked through the docs have not seen a option to achieve what I mentioned.

**Additional context**
N/A

**Comment by alex-courtis** [comment]
`update_root` is not quite the right place for this; it is for the current file.

1. This could be done with API: `:h nvim-tree-api.tree.change_root()`. We need a means to set `{path}` to the file's repository root.

Alternatives:

2. `tree.change_root({git_base = true})`
3. provide `git.base({path})` to pass to `change_root`

I'm leaning towards a recipe for 1.

**Comment by alex-courtis** [comment]
Have a play with this hacky mapping in your on_attach:

```lua
  vim.keymap.set("n", "x", function()
    local node = api.tree.get_node_under_cursor()

    local cmd = { "git", "-C", vim.fs.dirname(node.absolute_path), "rev-parse", "--show-toplevel" }

    -- strip trailing newline
    local toplevel = vim.fn.system(cmd):sub(0, -2)

    local stat, _ = vim.loop.fs_stat(toplevel)
    if stat and stat.type == "directory" then
      -- change to toplevel
      api.tree.change_root(toplevel)
    else
      -- failure message
      print(toplevel)
    end
  end, opts("CD git toplevel"))
```

**Comment by ckt114** [comment]
Hi @alex-courtis I tried your code and I do see that the `x` shows up as `CD git toplevel` but it doesn't work. I use [cljoly/telescope-repo.nvim](https://github.com/cljoly/telescope-repo.nvim) to open a file in another repo but when I opened nvim-tree it doesn't set that opened file's repo as root. What I'm trying to do is use only one neovim session and use telescope-repo to open another repo and wanting nvim-tree to automaticall set that newly opened repo as root.

**Comment by alex-courtis** [comment]
> Hi @alex-courtis I tried your code and I do see that the `x` shows up as `CD git toplevel` but it doesn't work.

What happens when you execute the mapping? Can you print `toplevel` to see what was retrieved?

> What I'm trying to do is use only one neovim session and use telescope-repo to open another repo and wanting nvim-tree to automaticall set that newly opened repo as root.

Baby steps. Let's get this going with a mapping then we can automate.
